### ISSUE TITLE
Add `list-gpg-user`command to list all added users

### ISSUE THREAD

**Issue by zhiboz** [AGWA/git-crypt#39]
Is it feasible to introduce a new command

    git-crypt list-gpg-user 

to list all added gpg users? Are there alternatives to show that info with the existing implementation?

**Comment by AGWA** [comment]
Yeah, this is planned.  For now, if you list the contents of the `.git-crypt/keys/default/0/` directory you'll the full GPG fingerprints of every authorized user.

**Comment by zhiboz** [comment]
Cool! Greatly appreciate your efforts!

**Comment by tachang** [comment]
They are the fingerprints and not the actual GPG keys?

**Comment by AGWA** [comment]
Yes, the filenames are the fingerprints, not the actual keys.

**Comment by tachang** [comment]
Gotcha. Is the .git-crypt directory actually encrypted as well? I don't see the need to encrypt it. Also wouldn't it make sense to store the public key as well? Or is it there and I am just missing it.

**Comment by AGWA** [comment]
The `.git-crypt` directory is not encrypted *by git-crypt* (in fact, git-crypt installs a `.gitattributes` file in `.git-crypt` to override any rules that might cause it to be encrypted inadvertently).

That said, each file inside `.git-crypt` is separately encrypted, *by GPG*, to the public key indicated by the fingerprint in the filename.  This is what allows contributors to unlock the repository using their GPG private key.

It wouldn't make sense for git-crypt to store the GPG public key.  That's what the GPG keyring is for.  Given a fingerprint, you can export the public key from GPG with `gpg --export`.

**Comment by hoangdangninh** [comment]
I'm waiting for this feature, this's gonna be great! 👍

**Comment by Aslan** [comment]
👍 for this subcommand

**Comment by Constantin07** [comment]
+1 very useful feature to have

**Comment by GregSharpe1** [comment]
I found this awesome blog post! https://www.devopsgroup.com/2016/12/08/easy-git-crypt-user-identification/

**Comment by Constantin07** [comment]
Thanks @GregSharpe1. Nice article!

**Comment by jakemorganDOG** [comment]
Wow @GregSharpe1, that's exacly what I was looking for!!

**Comment by chribsen** [comment]
Just check the git log
```
git log .git-crypt/
```

**Comment by bentterp** [comment]
```git log .git-crypt/keys/default/0/*.gpg```

**Comment by sobi3ch** [comment]
Create git alias to make pseudo-command `git crypt-users`
Add this..
```
[alias]
	crypt-users = ! git log  .git-crypt/keys/*/*/*.gpg | egrep '\\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}\\b'
```
to your `~/.gitconfig`

**Comment by jottr** [comment]
@AGWA is this still planned?
This feature would increase the UX and thus security by making it more transparent which users were granted access.

**Comment by AlexShmelev** [comment]
@AGWA Is there any updates on this issue ?

**Comment by ikegentz** [comment]
> Create git alias to make pseudo-command `git crypt-users` Add this..
> 
> ```
> [alias]
> 	crypt-users = ! git log  .git-crypt/keys/*/*/*.gpg | egrep '\\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}\\b'
> ```
> 
> to your `~/.gitconfig`
> 
> UPDATE: new version
> 
> ```
> [alias]
> 	crypt-users = "! git log --format=full .git-crypt/keys/*/*/*.gpg | grep -E '^            .+' | sed 's/ *$//g' | awk '{gsub(/^[ \t]+/,\"\"); print $0 }'"
> ```

If anybody is looking for one that only lists the emails (that was what I needed anyways):
```
[alias]
        crypt-users = "! git log --format=full .git-crypt/keys/*/*/*.gpg | grep -E '^            .+' | sed 's/ *$//g' | awk '{gsub(/^[ \t]+/,\"\"); print $0 }' | awk 'NF>1{print $NF}' | sed 's/.//;s/.$//'"
```

Wow, this worked like a charm. Thank you!
